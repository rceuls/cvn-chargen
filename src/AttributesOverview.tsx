import { CharacterAttributes } from './services/character.classes';
import { formatBonus, getModifier } from './services/util';

const attributeOrder: CharacterAttributes[] = [
  CharacterAttributes.STR,
  CharacterAttributes.DEX,
  CharacterAttributes.CON,
  CharacterAttributes.INT,
  CharacterAttributes.WIS,
  CharacterAttributes.CHA,
];

const AttributesOverview = ({
  attributes,
  calculateBonusXP,
}: {
  attributes: number[];
  calculateBonusXP: (input: number[]) => string;
}) => {
  return (
    <>
      <div className="row">
        <div className="col-sm">
          {attributeOrder.map((a: CharacterAttributes) => {
            return (
              <>
                <span className="font-weight-bold">
                  {CharacterAttributes[a]}
                </span>
                <span className="ml-2 mr-2">
                  {attributes[+a]} ({formatBonus(getModifier(attributes[+a]))})
                </span>
              </>
            );
          })}
        </div>
      </div>
      <div className="row">
        <div className="col-sm">
          <span className="font-weight-bold">Bonus XP</span>
          <span className="ml-2">{calculateBonusXP(attributes)}</span>
        </div>
      </div>
      <hr />
    </>
  );
};

export default AttributesOverview;
