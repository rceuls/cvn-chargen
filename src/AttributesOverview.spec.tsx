import renderer from 'react-test-renderer';
import AttributesOverview from './AttributesOverview';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AttributesOverview
        attributes={[10, 11, 12, 13, 14, 15]}
        calculateBonusXP={(_) => 'no'}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
