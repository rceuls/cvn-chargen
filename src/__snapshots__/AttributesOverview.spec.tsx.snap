// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        STR
      </span>
      <span
        className="ml-2 mr-2"
      >
        10
         (
        0
        )
      </span>
      <span
        className="font-weight-bold"
      >
        DEX
      </span>
      <span
        className="ml-2 mr-2"
      >
        11
         (
        0
        )
      </span>
      <span
        className="font-weight-bold"
      >
        CON
      </span>
      <span
        className="ml-2 mr-2"
      >
        12
         (
        0
        )
      </span>
      <span
        className="font-weight-bold"
      >
        INT
      </span>
      <span
        className="ml-2 mr-2"
      >
        13
         (
        +1
        )
      </span>
      <span
        className="font-weight-bold"
      >
        WIS
      </span>
      <span
        className="ml-2 mr-2"
      >
        14
         (
        +1
        )
      </span>
      <span
        className="font-weight-bold"
      >
        CHA
      </span>
      <span
        className="ml-2 mr-2"
      >
        15
         (
        +1
        )
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Bonus XP
      </span>
      <span
        className="ml-2"
      >
        no
      </span>
    </div>
  </div>,
  <hr />,
]
`;
